tapfreighter/chain_porter.go

```diff
@@ -451,6 +451,18 @@ func (p *ChainPorter) storeProofs(sendPkg *sendPackage) error {
 		serializedScriptKey := asset.ToSerialized(out.ScriptKey.PubKey)
 		sendPkg.FinalProofs[serializedScriptKey] = outputProof
 
+		// Before we import the proof into the proof archive, we'll
+		// validate it.
+		verifier := &proof.BaseVerifier{}
+		_, err = verifier.Verify(
```

Comment:
👍 

Should also help catch certain classes of bugs earlier. Like writing an invalid proof to disk, then it whens trying to post to a Universe.